src/clarity-angular/data/datagrid/render/main-renderer.ts

```diff
@@ -137,7 +137,22 @@ export class DatagridMainRenderer implements AfterContentInit, AfterViewChecked,
         this.shouldStabilizeColumns = false;
         if (this.columnsSizesStable) {
             // change in items might have introduced/taken away the scrollbar
-            this.organizer.scrollbar.next();
+
+            // FIXME: setTimeout is needed here because:
+            // When the user changes the page the following things happen:
+            // 1. The array which contains the items displayed is updated to contain the items on the new page.
+            // 2. An event is emitted which is subscribed to by the main renderer (this file) and this marks the
+            // shouldStabilizeColumns flag to true
+            // 3. While this is happening the datagrid is in the process of cleaning up the view. The view first
+            // renders the new displayed items and then cleans up the old items. But there is a point where the view
+            // contains the old items as well as the new items. So if the page size is 10 the view contains 20 items.
+            // This causes the datagrid body to overflow.
+            // Now since shouldStabilizeColumns was set to true, the scrollbar width is calculated
+            // and added to the datagrid header. Adding the setTimeout gives Angular time to clean up the view so that
+            // the scrollbar disappears.
```

Comment:
Can you just mention the fact that this only needs a timeout because of https://github.com/angular/angular/issues/19094?

That way we can remove it as soon as the issue is fixed on the Angular side.

Reply:
Done